src/network/networkd-link.c

```diff
@@ -1763,6 +1763,84 @@ bool link_has_carrier(Link *link) {
         return false;
 }
 
+static int link_address_genmode_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
+        int r;
+
+        assert(link);
+
+        if (IN_SET(link->state, LINK_STATE_FAILED, LINK_STATE_LINGER))
+                return 1;
+
+        r = sd_netlink_message_get_errno(m);
+        if (r < 0)
+                log_link_warning_errno(link, r, "Could not set address genmode for interface: %m");
+
+        return 1;
```

Comment:
What is the significance of `1` here? I guess none, so it should be 0 instead.

Reply:
I think while we process callback after we get a reply from kernel we return > 0 or 1 just to mention the success.